framework/validators/Validator.php

```diff
@@ -277,7 +277,7 @@ public function getValidationAttributes($attributes = null)
             return $this->getAttributeNames();
         }
 
-        if (is_string($attributes)) {
+        if (is_scalar($attributes)) {
```

Comment:
Why do you have int there? It's a list of attribute names according to phpdoc of both parent method and this method.

Reply:
@samdark Technically object in PHP can have property like `123`.

Reply:
We have generated the model and the form dynamically, so to have an unique identifier we have used the tables id - its not very common indeed, but it can happen though. Therefore just checking `if (!is_array($attributes))` would be safe as well.

Reply:
`!is_array` is true for objects and resources so it looks like a bad idea.

Reply:
Are `float` properties possible?

Reply:
I've put together a test case: https://3v4l.org/dQ5Yv

All the properties are strings...

Reply:
So yes, your fix could be applied, I think.

Reply:
> All the properties are strings...

`var_dump()` will report `string`, but integer properties will be cast to `int` if you try to export them as array: https://3v4l.org/4YZcV